controller/conditional_request.go

```diff
@@ -0,0 +1,329 @@
+	// HTTP header does not include microseconds, so we need to ignore them in the "updated_at" record field.
+	lastModified := entity.GetLastModified()
+	if !ctx.ModifiedSince(lastModified) {
+		return ctx.NotModified()
+	}
+	// check the ETag
+	eTag := entity.GenerateETag()
+	if ctx.MatchesETag(eTag) {
+		return ctx.NotModified()
+	}
+	// return the work item type along with conditional query and caching headers
+	ctx.SetHeader(LastModified, lastModified.String())
+	ctx.SetHeader(ETag, eTag)
+	ctx.SetHeader(CacheControl, MaxAge+"="+maxAge)
+	return ctx.OK(entity)
+}
+
+// WorkItemTypeSingle an alias to app.WorkItemTypeSingle
+type WorkItemTypeSingle app.WorkItemTypeSingle
+
+// GetLastModified gets the update time for a given element.
+func (entity WorkItemTypeSingle) GetLastModified() time.Time {
+	var updatedAt time.Time
+	if entity.Data.Attributes.UpdatedAt != nil {
+		updatedAt = entity.Data.Attributes.UpdatedAt.Truncate(time.Second).UTC()
+	}
+	return updatedAt
+}
+
+// GenerateETag generates the value to return in the "ETag" HTTP response header, using the data in the given buffer.
+// The ETag is the base64-encoded value of the md5 hash of the buffer content
+func (entity WorkItemTypeSingle) GenerateETag() string {
+	var buffer bytes.Buffer
+	// build a block of text for the given type with one <id>-<version>
+	buffer.WriteString(entity.Data.ID.String())
+	buffer.WriteString("-")
+	buffer.WriteString(strconv.Itoa(entity.Data.Attributes.Version))
+	buffer.WriteString("\n")
+	etagData := md5.Sum(buffer.Bytes())
+	etag := base64.StdEncoding.EncodeToString(etagData[:])
+	return etag
+}
+
+// ShowWorkitemtypeContext an alias to app.ShowWorkitemtypeContext
+type ShowWorkitemtypeContext struct {
+	app.ShowWorkitemtypeContext
+}
+
+// NotModified returns the `304 Not Modified` response with an empty body
+func (ctx ShowWorkitemtypeContext) NotModified() error {
+	return ctx.ShowWorkitemtypeContext.NotModified()
+}
+
+// OK returns the `200 OK` response with the given entity in the body
+func (ctx ShowWorkitemtypeContext) OK(entity interface{}) error {
+	res := app.WorkItemTypeSingle(entity.(WorkItemTypeSingle))
+	return ctx.ShowWorkitemtypeContext.OK(&res)
+}
+
+// ModifiedSince returns `true` if the context's `If-Modified-Since` header is `nil` or before the given `lastModified` argument.
+func (ctx ShowWorkitemtypeContext) ModifiedSince(lastModified time.Time) bool {
+	if ctx.IfModifiedSince != nil {
+		ifModifiedSince := ctx.IfModifiedSince.UTC()
+		return ifModifiedSince.Before(lastModified)
+	}
+	return true
+}
+
+// MatchesETag returns `true` the given `etag` matches with the context's `If-None-Match` header.
+func (ctx ShowWorkitemtypeContext) MatchesETag(etag string) bool {
+	if ctx.IfNoneMatch != nil && *ctx.IfNoneMatch == etag {
+		return true
+	}
+	return false
+}
+
+// SetHeader sets the header with the given name and value.
+func (ctx ShowWorkitemtypeContext) SetHeader(name, value string) {
```

Comment:
At first, I added those methods in order to access the fields of each context type. But indeed, we could have more explicit methods to set the `Last-Modified` and `ETag` headers.